tools/moFF/moff.xml

```diff
@@ -210,27 +223,50 @@
                 <param argument="--peptide_summary" type="boolean" value="true" label="Output the peptide summary?"/>
             </when>
             <when value="mbr">
                 <expand macro="ident_input_macro" allow_multiple="false" input_type="data_collection"/>
+                <param argument="--mbr" type="text" value="only" label="moFF workflow"
+                    help="select the moFF workflow" />
                 <param argument="--ext" type="text" value="tab" label="Provide the extension used in the display file name (without the period)"/>
             </when>
             <when value="all">
                 <expand macro="ident_input_macro" allow_multiple="false" input_type="data_collection"/>
                 <expand macro="raw_input_macro" allow_multiple="false" input_type="data_collection"/>
                 <param argument="--tol" type="float" value="10" label="Tolerance parameter"
                     help="Specify the tolerance parameter in ppm." />
-                <param argument="--rt_w" type="float" value="3.0" label="Retention time window"
+                <param argument="--mbr" type="text" value="on" label="moFF workflow"
+                    help="select the moFF workflow" />
+                <param argument="--xic_length" type="float" value="3.0" label="Retention time window"
                     help="Specify rt window for xic in minutes." />
-                <param argument="--rt_p" type="float" value="1" label="Time window for the peak"
+                <param argument="--rt_peak_win" type="float" value="1" label="Time window for the peak"
                     help="Specify the time windows for the peak in minutes." />
-                <param argument="--rt_p_match" type="float" value="1.2" label="Time window for the matched peak"
+                <param argument="--rt_peak_win" type="float" value="1.2" label="Time window for the matched peak"
                     help="Specify the time windows for the matched peak in minutes." />
+                <conditional name="match_filter">
+                    <param argument="match_filter_selector" type="boolean" label="Activate filtering of matched peptides" />
+                    <when value="true">
+                        <param name="--sample_size" label="sample_size" type="float" value="0.20" help="percentage of MS2 peptide used to estimated the threshold. Default value: 0.20" />
+						<param name="--quantile_thr_filtering" label="quantile_thr_filtering" type="float" value="0.75" help="quantile value used to compute the filtering threshold for the matched peak"/>
+						<conditional name="custom_ptm_file">
+							<param argument="custom_ptm_file_selector" type="boolean" label="use custom ptm file or internal moFF ptm file for PeptideShaker output" />
+					        <when value="true">
+						         <param name="--ptm_file " type="data" format="tabular" label="ptm_file" help="load your ptm file in order to overwrite internal method"/>
```

Comment:
> I'm a little bit puzzled why this conditional is needed. Do you just want to say that the --ptm_file is optional? If so please simply add `optional="True"` to the data-param and remove the conditional.
in case of filtering (match_filter is TRUE) --ptm_file must be specified: or a custom json file given by the user or the user can use the basi ptm_file.json that is located in moFF folder . How to do it ?

Reply:
Sorry, I don't understand the sentence.
As far as I understand you can do:

```
#if $ptm_file:
   --ptm_file '$task.ptm_file'
#else:
	--ptm_file ./ptm_setting_ps.json
#end if

...

<param argument="--ptm_file " type="data" format="tabular" optional="True" label="ptm_file" help="load your ptm file in order to overwrite internal method"/>
```

Reply:
fixed it ...